tests/sentry/search/events/test_builder.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+import unittest
+
+from django.utils import timezone
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import Condition, Op, Or
+from snuba_sdk.orderby import Direction, OrderBy
+
+from sentry.search.events.builder import QueryBuilder
+from sentry.utils.snuba import Dataset
+
+
+class QueryBuilderTest(unittest.TestCase):
+    def setUp(self):
+        self.start = datetime.datetime(2015, 5, 18, 10, 15, 1, tzinfo=timezone.utc)
+        self.end = datetime.datetime(2015, 5, 19, 10, 15, 1, tzinfo=timezone.utc)
+        self.projects = [1, 2, 3]
+        self.params = {
+            "project_id": self.projects,
+            "start": self.start,
+            "end": self.end,
+        }
+        # These conditions should always be on a query when self.params is passed
+        self.default_conditions = [
+            Condition(Column("timestamp"), Op.GTE, self.start),
+            Condition(Column("timestamp"), Op.LT, self.end),
+            Condition(Column("project_id"), Op.IN, self.projects),
+        ]
+
+    def test_simple_query(self):
+        query = QueryBuilder(
+            Dataset.Discover,
+            self.params,
+            "user.email:foo@example.com release:1.2.1",
+        )
+
+        assert query.where == [
+            Condition(Column("email"), Op.EQ, "foo@example.com"),
+            Condition(Column("release"), Op.EQ, "1.2.1"),
+            *self.default_conditions,
+        ]
```

Comment:
I would probably do `query.get_snql_query()` to make sure what you're building is actually valid.

Reply:
:+1: that makes sense, should I do `query.get_snql_query().validate()` too? Or is instantiating the `Query` sufficient?

Reply:
nvm answered my own q, gotta call the validate too, the super simple tests fail cause of 
```
E   snuba_sdk.query_visitors.InvalidQuery: query must have at least one expression in select
```